src/core/storage/ionicStorage/ionicStorage.ts

```diff
@@ -87,7 +87,7 @@ class IonicStorage<T extends BaseRecord> implements StorageService<T> {
     this.checkSession(this.session);
     const recordStorage = await this.session!.get(id);
 
-    if (!recordStorage) {
+    if (!recordStorage || recordStorage.category !== recordClass.type) {
```

Comment:
hmmm, if we can check here before `deserializeRecord` - we should just do this and not change `deserializeRecord` in this PR.

For the others, you can just skip pushing into `instances` if it doesn't match. This would keep `deserializeRecord` cleaner and also be more efficient since we check before deserialising

Reply:
But this has to be done for SQLite too - not sure if you can do this before deserialising for that - let me know